### ISSUE TITLE
ISC003 false positive

### ISSUE THREAD

**Issue by trim21** [charliermarsh/ruff#3934]
ruff 0.0.261

`pyproject.toml`
```toml

[tool.ruff]
select = [
 "ISC"
]
```

```python
RED = ""
COLOR_END = ""

DATA_LOSS_MESSAGE = RED + "All Data will lost, please backup first" + COLOR_END


def update_database() -> None:
    print(
        (
            "s1 "
            + "s2\n"
            + DATA_LOSS_MESSAGE
        )
    )

```

app.py:9:9: ISC003 Explicitly concatenated string should be implicitly concatenated

you can't implicitly concat string literial and variable

**Comment by charliermarsh** [comment]
Oh, good call!

**Comment by trim21** [comment]
> I think it's fine as it's complaining only when both the left and right are string literals:
> 
> ```python
> def update_database() -> None:
>     print(
>         (
>             "s1 "  # <-- it's compaining here
>             + "s2\n"
>           # ^ when you remove this `+`, it stops complaining as expected
>             + DATA_LOSS_MESSAGE
>         )
>     )
> ```
> 
> Is the expectation that if there's at least one variable then the warning shouldn't be raised?

sounds reasonable

**Comment by charliermarsh** [comment]
Ah yeah, I guess it's arguably not a _bug_, but more a matter of what's most sensible.

**Comment by trim21** [comment]
in that case should i close it? Or you want to re-consider this rule and leave it open?

**Comment by charliermarsh** [comment]
On further reflection, let's close for now. I think it would arguably be surprising _not_ to flag this.
